sql/src/main/java/io/crate/executor/transport/TransportShardAction.java

```diff
@@ -143,9 +141,21 @@ public void killJob(UUID jobId) {
         }
     }
 
-    protected abstract Tuple<ShardResponse, Translog.Location> processRequestItems(ShardId shardId,
-                                                                                   Request request,
-                                                                                   AtomicBoolean killed) throws InterruptedException;
+    /**
+     * Wrapper of the {@link #processRequestItems(ShardId, ShardRequest, AtomicBoolean)} to facilitate testing as
+     * the WritePrimaryResult is not exposed
+     */
+    @VisibleForTesting
+    Tuple<ShardResponse, Translog.Location> processRequestToTuple(ShardId shardId,
```

Comment:
looks odd to declare a method here just for testing. don't get why WritePrimaryResult should not be available in an unit test in the same package..